tests/conftest.py

```diff
@@ -450,21 +451,30 @@ def aws_credentials(monkeypatch):
 
 
 @pytest.fixture(scope="function")
-def mock_s3_resource(aws_credentials):
+def mock_s3_resource(aws_credentials, mock_s3_client):
     if moto_server_address is None:
         with moto.mock_aws():
             yield boto3.resource("s3", region_name=TEST_AWS_REGION_NAME)
     else:
         yield boto3.resource("s3", region_name=TEST_AWS_REGION_NAME, endpoint_url=moto_server_address)
 
 @pytest.fixture(scope="function")
-def mock_s3_client(aws_credentials):
+def get_mock_s3_client(aws_credentials):
```

Comment:
It's confusing to change the name of this fixture with a `get_` prefix:  
when using it in a test `get_mock_s3_client` will have the value of a client, not a getter of clients